tasks/Module 3/task_1.py

```diff
@@ -0,0 +1,59 @@
+"""Task 1 of Module 3."""
+import pyinputplus as pyip
+
+BREAD = {'wheat': 1.321, 'white': 2.5, 'sourdough': 40}
+PROTEIN = {'chicken': 32.3, 'turkey': 55, 'ham': 1, 'tofu': 0.1}
+CHEESE = {'cheddar': 5, 'Swiss': 5.5, 'mozzarella': 1000}
+SAUCE = {'mayo': 300, 'mustard': 400, 'lettuce': 500, 'tomato': 600}
+
+sandwich_ingredients = []
+
+
+def sandwich_maker():
+    """Provide information on ordered sandwich/es.
+
+    Based on users input, provides information on:
+    number of ordered sandwiches;
+    sandwich's ingredients;
+    order's price
+    """
+    while True:
+
+        bread_type = pyip.inputMenu(['wheat', 'white', 'sourdough'])
```

Comment:
I would extract all choices into constants just like you did with prices. Here you can introduce constant like `BREAD_TYPE_CHOICES` or something similar.

Reply:
Hi @Wiktor
first i thought to create new costants for bread etc, but i tought "why not to use what we already have?" and wanted to use all keys in BREAD as BREAD.keys. Unfortunatelly this throwing an error:

`File "E:/Basics-of-Python/tasks/Module 3/task_1.py", line 21, in sandwich_maker
    bread_type = pyip.inputMenu(BREAD.keys())
  File "E:\Basics-of-Python\venv\lib\site-packages\pyinputplus\__init__.py", line 747, in inputMenu
    pysv._validateParamsFor_validateChoice(
  File "E:\Basics-of-Python\venv\lib\site-packages\pysimplevalidate\__init__.py", line 707, in _validateParamsFor_validateChoice
    raise PySimpleValidateException("choices arg must be a sequence")
pysimplevalidate.PySimpleValidateException: choices arg must be a sequence`

is there a way to get key from BREAD dict and use it in inputMenu, or it's better to have 4 more constants for every ingredient

Reply:
I think parsing it to the list will help. Like `bread_type = pyip.inputMenu(list(BREAD.keys()))`

Reply:
@GeorgiiDermenzhy it is actually very good idea to re-use existing constants, much better than introducing new ones 👍 

The error message `pysimplevalidate.PySimpleValidateException: choices arg must be a sequence` may look cryptic at the beginning, but basically sequences are collections`*` that are ordered and reversible: you can access element at arbitrary position by `sequence[some_int]` and you can use well... `reversed` function on it. Built-in sequences in you knowPython are `list`, `tuple` and `str`. Others like `dict` or `set` are collections but not sequences. Both sequences and collections are "iterable" which means you can use them in for loop. 

**Warning: more technical details below**
In addition object returned by `dict.keys()` is not regular Python collection you know but instance of special `dict_keys`. It has special abilities eg. it knows about changes in your dictionary:
```
>>> d = {'a': 1}
>>> k = d.keys()
>>> k
dict_keys(['a'])
>>> d['b'] = 2
>>> k
dict_keys(['a', 'b'])
```
But it is not a sequence:
```
>>> k[0]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_keys' object is not subscriptable
>>> reversed(k)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'dict_keys' object is not reversible
```

`*` You may now wondering what is exact definition of collection :) Basically anything that supports all of following techniques:

- `for something in collection:`
- `len(collection)`
- `something in collection`

So you can do all these operation with all object from this comment, but only with sequences you can do:
- `sequence[3]`
- `reversed(sequence)

Reply:
Hi Guys,
Thanks for the notes :)
@Filip, parse it to list worked perfect.
@Wiktor, I was reading it and re-reading xD don't know, if I got this(most probably not), but per my understanding `InputMenu` is awaiting for a `list/tuple` (that is a sequence) to be provided as an argument, but instead of it, it got `collection` that is not a `sequence`. 
And sequence is required because ....I don't know, could "list" all the keys, as `string` or `set` of strings? So, when I was using `BREAD.keys` it was trying to provide an address of `dict_keys object`. And this `dict_keys` is an object but not a sequence?

Reply:
@GeorgiiDermenzhy you're almost right. I think the key point here is that if something expects sequence of strings, you can give it either list or tuple.  When you provided `BREAD.keys()`, you provided `dict_keys` object that is collection but not sequence.

Reply:
All inputMenu has been changed to use keys of CONSTs